### DISCUSSION TITLE
Mapping profiles

### DISCUSSION THREAD

**Discussion by alicicek1** [Biarity/Sieve#149] [General]
Is there any way to map all properties instead o line by line for all entities?

```protected override SievePropertyMapper MapProperties(SievePropertyMapper mapper)
protected override SievePropertyMapper MapProperties(SievePropertyMapper mapper)
        {
            mapper.Property<SurveyUserEntity>(a => a.Survey.Type)
                .CanSort()
                .CanFilter();
            mapper.Property<SurveyUserEntity>(a => a.CreatedByUser.SchoolId)
                .CanSort()
                .CanFilter();

            return mapper;
        }
`

**Comment by hasanmanzak** [comment]
This would require to walk through object structure that would bring whole all lot of overhead. And way more edge case handling: what would happen with cyclic dependencies? That brings "CyclicDependencyHandler" or "...Policy" etc. Indecisive, I am..

**Reply by alicicek1** [reply]
Yes can be  but can't sieve realize which model I have been filtering for? For ex, I have CreatedUser in my base entity related to the UserEntity and I send request to a service , those services methods take a Model as a parameter and should I map all entities for filtering for CreatedUser.Name?

**Comment by maskalek** [comment]
Also don't like this behaviour that you have to explicitly mark all properties you need. There should be a method "CanSortAll" or a default setting.
